src/utils.cpp

```diff
@@ -42,12 +42,25 @@ std::string sysError() noexcept
     return strerror(errno);
 }
 
-Utils::EVPKeyPtr readPublicKey(const std::string& fileName)
+Utils::EVPKeyPtr readPublicKey(const std::string& src)
 {
-    FilePtr fp(fopen(fileName.c_str(), "rb"));
-    if (!fp)
-        throw JWTXX::Key::Error("Can't open key file '" + fileName + "'. " + sysError());
-    return Utils::EVPKeyPtr(PEM_read_PUBKEY(fp.get(), nullptr, nullptr, nullptr));
+    FilePtr fp(fopen(src.c_str(), "rb"));
```

Comment:
we could check whether the string starts with `-----BEGIN` and if so then we can straightway try to read pubkey from the base64. If it doesn't start with then we can try to open it as file.

Reply:
@sivachandran It is not our case, we pass raw string without `BEGIN/END`
See test https://github.com/RealImage/jwtxx/pull/1/files#diff-c221b935bd4506307c4782e4f01b441eR26